qa/L0_backend_python/python_unittest.py

```diff
@@ -48,6 +51,17 @@ def test_python_unittest(self):
             # Otherwise, it will return 0.
             self.assertEqual(output0, [1])
 
+    def test_python_unittest(self):
+        model_name = os.environ['MODEL_NAME']
+
+        # The shared memory size changes during these tests and as a result
```

Comment:
Can we provide an optional "threshold" to the probe? This way we can test whether the increment or decrement in the shared memory regions is within expected range.

Anyways you should elaborate more here. A little more about why is the memory change expected. Is it an increase or decrease?

Reply:
Also, does it keep increasing with every infer call? If it is only for once- initialization, then we can run the unittest again within probe this time after L60.

Reply:
> Can we provide an optional "threshold" to the probe? This way we can test whether the increment or decrement in the shared memory regions is within expected range.

The problem here is that the shared memory region might grow as a result of requiring more memory. Because of this when the shared memory probe is examining the memory usage, it might discover that the free memory has increased instead of being the same. Per your second suggestion, I called the tests that grow the region two times and did the shared memory probing in the second call.

Reply:
> Per your second suggestion, I called the tests that **grow the region two times** and did the shared memory probing in the second call.

If the test is growing the memory even in second invocation then wouldn't the test still fail along the similar reasoning?

Reply:
It wouldn't grow for the second time because now the region is grown and doesn't require additional memory. Let's say the inference requires 128MB of shared memory and the region is 64 MB. When the inference is called, the region is grown to 128MB. When the second inference happens, the region size is still 128 and there is no need for growing the region.